File: Arrays and String/209. Minimum Size Subarray Sum/min_subarray.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        r = -1
        sum = 0
        min_len = len(nums) + 1

        while l < len(nums):
            if r + 1 < len(nums) and sum < target:
                r += 1
                sum += nums[r]
            else:
                sum -= nums[l]
                l += 1
            if sum >= target:
                min_len = min(min_len, r - l + 1)

        min_len = 0 if min_len == len(nums) + 1 else min_len
        return min_len

File: Arrays and String/209. Minimum Size Subarray Sum/min_subarray.cpp
class Solution
{
public:
    int minSubArrayLen(int target, vector<int> &nums)
    {
        int l = 0;
        int r = -1;
        int sum = 0;
        int min_len = nums.size() + 1;

        while (l < nums.size())
        {
            if (r + 1 < nums.size() && sum < target)
            {
                sum += nums[++r];
            }
            else
            {
                sum -= nums[l++];
            }
            if (sum >= target)
            {
                min_len = min(min_len, r - l + 1);
            }
        }
        min_len = min_len == nums.size() + 1 ? 0 : min_len;
        return min_len;
    }
};